bokeh/sphinxext/bokeh_gallery.py

```diff
@@ -98,19 +98,25 @@ def config_inited_handler(app, config):
     # left at the end (and remove it in that case)
     extras = set(os.listdir(gallery_dir))
 
-    # app.env.note_dependency(specpath)
-    spec = json.load(open(gallery_file))
-    details = spec["details"]
-
-    names = {x["name"] for x in details}
-    if len(names) < len(details):
-        raise SphinxError(f"gallery file {gallery_file!r} has duplicate names")
+    basepath = app.srcdir.replace('/sphinx/source', '')
+    folder = ['plotting','models']
+    details = []
+    detail_names = []
+    for f in folder:
+        path = f'examples/{f}/file'
+        for name in os.listdir(join(basepath, path)):
+            if not name.startswith('_') and name.endswith('.py') and  not 'custom_layout' in name:
```

Comment:
it's probably good to have and "exclusion list" capability, but let's make it properly configurable. For a simple start, a set `EXCLUDES` that has the "full" explicit path `examples/plotting/file/custom_layout.py` is probably the best in case there are filename duplicates in different directories.